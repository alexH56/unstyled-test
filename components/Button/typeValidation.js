import PropTypes from 'prop-types';

export const buttonPropTypes = {
    // Type/color of button
    type: PropTypes.oneOf(['primary', 'secondary', 'tertiary', 'dark', 'light', 'success', 'warning', 'danger', 'text']),
    // How large should the button be?
    size: PropTypes.oneOf(['sm', 'md', 'lg', 'xl']),
    // Filled or ghost
    ghost: PropTypes.bool,    
    // if true, button has loading animation
    processing: PropTypes.bool,
    // if true, button is disabled and opacity is reduced
    disabled: PropTypes.bool,
    // Optional, additional styling
    className: PropTypes.string,
    // Optional click handler  
    onClick: PropTypes.func,
    // boolean options available for type:
    primary: PropTypes.bool,
    secondary: PropTypes.bool,
    tertiary: PropTypes.bool,
    dark: PropTypes.bool,
    light: PropTypes.bool,
    success: PropTypes.bool,
    warning: PropTypes.bool,
    danger: PropTypes.bool,
    text: PropTypes.bool,
    // boolean options available for size:
    sm: PropTypes.bool,
    md: PropTypes.bool,
    lg: PropTypes.bool,
    xl: PropTypes.bool
  };