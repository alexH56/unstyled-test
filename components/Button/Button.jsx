import React from 'react'
import cx from '../../utils/classes'
import { buttonPropTypes } from './typeValidation'
import { baseStyles, buttonSizes, buttonTypes } from './styles/buttonStyles'

export default function Button({
    primary = false,
    secondary = false,
    tertiary = false,
    dark = false,
    light = false,
    success = false,
    warning = false,
    danger = false,
    text = false,
    ghost = false,
    sm = false,
    md = false,
    lg = false,
    xl = false,
    iconLeft,
    iconRight,
    onClick,
    processing,
    disabled,
    className,
    children = 'Hello',
    type,
    size,
}) {
    const types = {
        primary,
        secondary,
        tertiary,
        dark,
        light,
        success,
        warning,
        danger,
        text,
    }
    const givenType =
        Object.keys(types).filter((t) => types[t])[0] || type || 'primary'

    const sizes = { sm, md, lg, xl }
    const givenSize =
        Object.keys(sizes).filter((t) => sizes[t])[0] || size || 'sm'

    return (
        <button
            disabled={disabled}
            onClick={onClick}
            className={cx(
                baseStyles,
                buttonSizes[givenSize],
                givenType === 'text'
                    ? buttonTypes.text
                    : buttonTypes[givenType][!ghost ? 'filled' : 'ghost'],
                disabled ? 'disabled:opacity-40 cursor-not-allowed' : '',
                className
            )}
        >
            {processing ? (
                <svg
                    xmlns="http://www.w3.org/2000/svg"
                    className="h-6 w-6 animate-pulse"
                    fill="none"
                    viewBox="0 0 24 24"
                    stroke="currentColor"
                >
                    <path
                        strokeLinecap="round"
                        strokeLinejoin="round"
                        strokeWidth={2}
                        d="M5 12h.01M12 12h.01M19 12h.01M6 12a1 1 0 11-2 0 1 1 0 012 0zm7 0a1 1 0 11-2 0 1 1 0 012 0zm7 0a1 1 0 11-2 0 1 1 0 012 0z"
                    />
                </svg>
            ) : (
                <>
                    {iconLeft && <span>{iconLeft}</span>}
                    <span>{children}</span>
                    {iconRight && <span>{iconRight}</span>}
                </>
            )}
        </button>
    )
}

Button.propTypes = {
    ...buttonPropTypes,
}
