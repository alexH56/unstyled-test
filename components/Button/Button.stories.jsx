import React from 'react'
import Button from './Button.jsx'
import '../../src/index.css'

export default {
    title: 'Untitled UI/Atoms/Button',
    component: Button,
    argTypes: {
        type: {
            options: [
                'primary',
                'secondary',
                'tertiary',
                'dark',
                'light',
                'success',
                'warning',
                'danger',
                'text',
            ],
            control: { type: 'select' },
        },
        size: {
            options: ['sm', 'md', 'lg', 'xl'],
            control: { type: 'radio' },
        },
        className: {
            control: { type: 'text' },
        },
        primary: {
            table: { disable: true },
        },
        secondary: {
            table: { disable: true },
        },
        tertiary: {
            table: { disable: true },
        },
        dark: {
            table: { disable: true },
        },
        light: {
            table: { disable: true },
        },
        success: {
            table: { disable: true },
        },
        warning: {
            table: { disable: true },
        },
        danger: {
            table: { disable: true },
        },
        text: {
            table: { disable: true },
        },
        sm: {
            table: { disable: true },
        },
        md: {
            table: { disable: true },
        },
        lg: {
            table: { disable: true },
        },
        xl: {
            table: { disable: true },
        },
        baseStyles: {
            table: { disable: true },
        },
        onClick: {
            action: 'clicked',
            table: { disable: true },
        },
    },
}

const Template = (args) => <Button {...args} />

export const Primary = Template.bind({})
Primary.args = {
    type: 'primary',
    children: 'Primary',
    ghost: false,
    size: 'md',
    disabled: false,
    processing: false,
}

export const Alt = Template.bind({})
Alt.args = {
    type: 'primary',
    children: 'Alt',
    ghost: true,
    disabled: false,
    processing: false,
}

export const CTA = Template.bind({})
CTA.args = {
    type: 'primary',
    children: 'CTA',
    size: 'xl',
    ghost: false,
    processing: false,
    disabled: false,
}

export const Loading = Template.bind({})
Loading.args = {
    type: 'primary',
    ghost: false,
    processing: true,
    disabled: false,
}

export const Success = Template.bind({})
Success.args = {
    type: 'success',
    children: 'Success!',
    ghost: false,
    processing: false,
    disabled: false,
}

export const Warning = Template.bind({})
Warning.args = {
    type: 'warning',
    children: 'Warning!',
    ghost: false,
    processing: false,
    disabled: false,
}

export const Danger = Template.bind({})
Danger.args = {
    type: 'danger',
    children: 'Danger!',
    ghost: false,
    processing: false,
    disabled: false,
}
