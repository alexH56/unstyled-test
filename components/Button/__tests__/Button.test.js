import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button.js';

  test('Primary Button Snapshot', () => {
    const tree = renderer
      .create(<Button primary>Primary Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Primary Ghost Button Snapshot', () => {
    const tree = renderer
      .create(<Button type="primary" ghost>Primary Ghost Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Secondary Button Snapshot', () => {
    const tree = renderer
      .create(<Button secondary>Secondary Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Secondary Ghost Button Snapshot', () => {
    const tree = renderer
      .create(<Button type="secondary" ghost>Secondary Ghost Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Tertiary Button Snapshot', () => {
    const tree = renderer
      .create(<Button tertiary>Tertiary Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Tertiary Ghost Button Snapshot', () => {
    const tree = renderer
      .create(<Button type="tertiary" ghost>Tertiary Ghost Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Dark Button Snapshot', () => {
    const tree = renderer
      .create(<Button dark>Dark Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Dark Ghost Button Snapshot', () => {
    const tree = renderer
      .create(<Button type="dark" ghost>Dark Ghost Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Light Button Snapshot', () => {
    const tree = renderer
      .create(<Button light>Light Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Light Ghost Button Snapshot', () => {
    const tree = renderer
      .create(<Button type="light" ghost>Light Ghost Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Success Button Snapshot', () => {
    const tree = renderer
      .create(<Button success>Success Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Success Ghost Button Snapshot', () => {
    const tree = renderer
      .create(<Button type="success" ghost>Success Ghost Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Warning Button Snapshot', () => {
    const tree = renderer
      .create(<Button warning>Warning Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Warning Ghost Button Snapshot', () => {
    const tree = renderer
      .create(<Button type="warning" ghost>Warning Ghost Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Danger Button Snapshot', () => {
    const tree = renderer
      .create(<Button danger>Danger Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Danger Ghost Button Snapshot', () => {
    const tree = renderer
      .create(<Button type="danger" ghost>Danger Ghost Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Text Button Snapshot', () => {
    const tree = renderer
      .create(<Button text>Text Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Small Button Snapshot', () => {
    const tree = renderer
      .create(<Button size="sm">Small Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Large Button Snapshot', () => {
    const tree = renderer
      .create(<Button lg>Large Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Xtra Large Button Snapshot', () => {
    const tree = renderer
      .create(<Button size="xl">Xtra Large Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Disabled Button Snapshot', () => {
    const tree = renderer
      .create(<Button disabled>Disabled Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Icon Left Button Snapshot', () => {
    const tree = renderer
      .create(<Button iconLeft="ICON">Left Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Icon Right Button Snapshot', () => {
    const tree = renderer
      .create(<Button iconRight="ICON">Right Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Processing Button Snapshot', () => {
    const tree = renderer
      .create(<Button type="primary" processing>Processing Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Default Props Button Snapshot', () => {
    const tree = renderer
      .create(<Button></Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
