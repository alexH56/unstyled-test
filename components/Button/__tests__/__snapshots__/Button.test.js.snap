// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Danger Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border border-red-500 bg-red-500 text-white hover:text-red-500 hover:bg-opacity-30 hover:border-red-500 dark:bg-opacity-30 dark:hover:text-white dark:hover:bg-opacity-70 disabled:border-red-500 disabled:bg-red-500 disabled:text-white"
>
  <span>
    Danger Button
  </span>
</button>
`;

exports[`Danger Ghost Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border text-red-500 bg-transparent border-red-500 hover:text-red-500 hover:bg-red-500 hover:bg-opacity-30 dark:text-white dark:hover:text-white disabled:text-red-500 disabled:bg-transparent disabled:border-red-500"
>
  <span>
    Danger Ghost Button
  </span>
</button>
`;

exports[`Dark Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border border-dark-900 bg-dark-900 text-white hover:text-dark-700 hover:bg-dark-200 hover:border-dark-500 dark:bg-dark-300 dark:text-light dark:hover:bg-dark-700 disabled:border-dark-900 disabled:bg-dark-900 disabled:text-white"
>
  <span>
    Dark Button
  </span>
</button>
`;

exports[`Dark Ghost Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border text-dark-700 bg-transparent border-dark-500 hover:border-dark-900 hover:bg-dark-200 hover:text-dark-700 dark:text-light dark:border-dark-700 dark:hover:bg-dark-300 dark:hover:text-light disabled:text-dark-900 disabled:bg-transparent disabled:border-dark-900 dark:disabled:border-dark-100 dark:disabled:text-dark-100 dark:disabled:bg-transparent"
>
  <span>
    Dark Ghost Button
  </span>
</button>
`;

exports[`Default Props Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border border-primary-900 bg-primary-900 text-white hover:text-primary-700 hover:bg-primary-200 hover:border-primary-500 dark:bg-primary-300 dark:text-light dark:hover:bg-primary-700 disabled:border-primary-900 disabled:bg-primary-900 disabled:text-white"
>
  <span>
    Hello
  </span>
</button>
`;

exports[`Disabled Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border border-primary-900 bg-primary-900 text-white hover:text-primary-700 hover:bg-primary-200 hover:border-primary-500 dark:bg-primary-300 dark:text-light dark:hover:bg-primary-700 disabled:border-primary-900 disabled:bg-primary-900 disabled:text-white disabled:opacity-40 cursor-not-allowed"
  disabled={true}
>
  <span>
    Disabled Button
  </span>
</button>
`;

exports[`Icon Left Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border border-primary-900 bg-primary-900 text-white hover:text-primary-700 hover:bg-primary-200 hover:border-primary-500 dark:bg-primary-300 dark:text-light dark:hover:bg-primary-700 disabled:border-primary-900 disabled:bg-primary-900 disabled:text-white"
>
  <span>
    ICON
  </span>
  <span>
    Left Button
  </span>
</button>
`;

exports[`Icon Right Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border border-primary-900 bg-primary-900 text-white hover:text-primary-700 hover:bg-primary-200 hover:border-primary-500 dark:bg-primary-300 dark:text-light dark:hover:bg-primary-700 disabled:border-primary-900 disabled:bg-primary-900 disabled:text-white"
>
  <span>
    Right Button
  </span>
  <span>
    ICON
  </span>
</button>
`;

exports[`Large Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-lg px-8 py-1 justify-center border border-primary-900 bg-primary-900 text-white hover:text-primary-700 hover:bg-primary-200 hover:border-primary-500 dark:bg-primary-300 dark:text-light dark:hover:bg-primary-700 disabled:border-primary-900 disabled:bg-primary-900 disabled:text-white"
>
  <span>
    Large Button
  </span>
</button>
`;

exports[`Light Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border border-light-900 bg-light-900 text-white hover:text-light-700 hover:bg-light-200 hover:border-light-500 dark:bg-light-300 dark:text-light dark:hover:bg-light-700 disabled:border-light-900 disabled:bg-light-900 disabled:text-white"
>
  <span>
    Light Button
  </span>
</button>
`;

exports[`Light Ghost Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border text-light-700 bg-transparent border-light-500 hover:border-light-900 hover:bg-light-200 hover:text-light-700 dark:text-light dark:border-light-700 dark:hover:bg-light-300 dark:hover:text-light disabled:text-light-900 disabled:bg-transparent disabled:border-light-900 dark:disabled:border-light-100 dark:disabled:text-light-100 dark:disabled:bg-transparent"
>
  <span>
    Light Ghost Button
  </span>
</button>
`;

exports[`Primary Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border border-primary-900 bg-primary-900 text-white hover:text-primary-700 hover:bg-primary-200 hover:border-primary-500 dark:bg-primary-300 dark:text-light dark:hover:bg-primary-700 disabled:border-primary-900 disabled:bg-primary-900 disabled:text-white"
>
  <span>
    Primary Button
  </span>
</button>
`;

exports[`Primary Ghost Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border text-primary-700 bg-transparent border-primary-500 hover:border-primary-900 hover:bg-primary-200 hover:text-primary-700 dark:text-light dark:border-primary-700 dark:hover:bg-primary-300 dark:hover:text-light disabled:text-primary-900 disabled:bg-transparent disabled:border-primary-900 dark:disabled:border-primary-100 dark:disabled:text-primary-100 dark:disabled:bg-transparent"
>
  <span>
    Primary Ghost Button
  </span>
</button>
`;

exports[`Processing Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border border-primary-900 bg-primary-900 text-white hover:text-primary-700 hover:bg-primary-200 hover:border-primary-500 dark:bg-primary-300 dark:text-light dark:hover:bg-primary-700 disabled:border-primary-900 disabled:bg-primary-900 disabled:text-white"
>
  <svg
    className="h-6 w-6 animate-pulse"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M5 12h.01M12 12h.01M19 12h.01M6 12a1 1 0 11-2 0 1 1 0 012 0zm7 0a1 1 0 11-2 0 1 1 0 012 0zm7 0a1 1 0 11-2 0 1 1 0 012 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
    />
  </svg>
</button>
`;

exports[`Secondary Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border border-secondary-900 bg-secondary-900 text-white hover:text-secondary-700 hover:bg-secondary-200 hover:border-secondary-500 dark:bg-secondary-300 dark:text-light dark:hover:bg-secondary-700 disabled:border-secondary-900 disabled:bg-secondary-900 disabled:text-white"
>
  <span>
    Secondary Button
  </span>
</button>
`;

exports[`Secondary Ghost Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border text-secondary-700 bg-transparent border-secondary-500 hover:border-secondary-900 hover:bg-secondary-200 hover:text-secondary-700 dark:text-light dark:border-secondary-700 dark:hover:bg-secondary-300 dark:hover:text-light disabled:text-secondary-900 disabled:bg-transparent disabled:border-secondary-900 dark:disabled:border-secondary-100 dark:disabled:text-secondary-100 dark:disabled:bg-transparent"
>
  <span>
    Secondary Ghost Button
  </span>
</button>
`;

exports[`Small Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border border-primary-900 bg-primary-900 text-white hover:text-primary-700 hover:bg-primary-200 hover:border-primary-500 dark:bg-primary-300 dark:text-light dark:hover:bg-primary-700 disabled:border-primary-900 disabled:bg-primary-900 disabled:text-white"
>
  <span>
    Small Button
  </span>
</button>
`;

exports[`Success Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border border-green-500 bg-green-500 text-white hover:text-green-500 hover:bg-opacity-30 hover:border-green-500 dark:bg-opacity-30 dark:hover:text-white dark:hover:bg-opacity-70 disabled:border-green-500 disabled:bg-green-500 disabled:text-white"
>
  <span>
    Success Button
  </span>
</button>
`;

exports[`Success Ghost Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border text-green-500 bg-transparent border-green-500 hover:text-green-500 hover:bg-green-500 hover:bg-opacity-30 dark:text-white dark:hover:text-white disabled:text-green-500 disabled:bg-transparent disabled:border-green-500"
>
  <span>
    Success Ghost Button
  </span>
</button>
`;

exports[`Tertiary Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border border-tertiary-900 bg-tertiary-900 text-white hover:text-tertiary-700 hover:bg-tertiary-200 hover:border-tertiary-500 dark:bg-tertiary-300 dark:text-light dark:hover:bg-tertiary-700 disabled:border-tertiary-900 disabled:bg-tertiary-900 disabled:text-white"
>
  <span>
    Tertiary Button
  </span>
</button>
`;

exports[`Tertiary Ghost Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border text-tertiary-700 bg-transparent border-tertiary-500 hover:border-tertiary-900 hover:bg-tertiary-200 hover:text-tertiary-700 dark:text-light dark:border-tertiary-700 dark:hover:bg-tertiary-300 dark:hover:text-light disabled:text-tertiary-900 disabled:bg-transparent disabled:border-tertiary-900 dark:disabled:border-tertiary-100 dark:disabled:text-tertiary-100 dark:disabled:bg-transparent"
>
  <span>
    Tertiary Ghost Button
  </span>
</button>
`;

exports[`Text Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 hover:text-primary-900 hover:bg-primary-200 hover:bg-opacity-80 dark:border dark:border-transparent dark:text-light dark:hover:border-primary-700 disabled:text-primary-3000 disabled:bg-transparent dark:disabled:text-primary-400 dark:disabled:bg-transparent"
>
  <span>
    Text Button
  </span>
</button>
`;

exports[`Warning Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border border-yellow-500 bg-yellow-500 text-white hover:text-yellow-500 hover:bg-opacity-30 hover:border-yellow-500 dark:bg-opacity-30 dark:hover:text-white dark:hover:bg-opacity-70 disabled:border-yellow-500 disabled:bg-yellow-500 disabled:text-white"
>
  <span>
    Warning Button
  </span>
</button>
`;

exports[`Warning Ghost Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-sm px-4 py-1 justify-center border text-yellow-500 bg-transparent border-yellow-500 hover:text-yellow-500 hover:bg-yellow-500 hover:bg-opacity-30 dark:text-white dark:hover:text-white disabled:text-yellow-500 disabled:bg-transparent disabled:border-yellow-500"
>
  <span>
    Warning Ghost Button
  </span>
</button>
`;

exports[`Xtra Large Button Snapshot 1`] = `
<button
  className="inline-flex space-x-2 items-center align-center rounded-sm transition-colors text-2xl px-10 py-1.5 justify-center border border-primary-900 bg-primary-900 text-white hover:text-primary-700 hover:bg-primary-200 hover:border-primary-500 dark:bg-primary-300 dark:text-light dark:hover:bg-primary-700 disabled:border-primary-900 disabled:bg-primary-900 disabled:text-white"
>
  <span>
    Xtra Large Button
  </span>
</button>
`;
